[15기 하태현] FormData 타입으로 변경(해결 완료), 리뷰 반영 완료

HTHstudy opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!



<img width="359" alt="스크린샷 2021-01-03 오후 11 40 55" src="https://user-images.githubusercontent.com/58502653/103481259-26636b80-4e1d-11eb-815f-f8a54a75b5a1.png">

## 수정 사항 간략한 한줄 요약
- 글쓰기 페이지 레이아웃 완성
- 기능 구현 완료
-  카테고리 선택 페이지 완료


		 
## 수정 사항들 자세한 내용
- 휴대폰 앨범에 접근해 사진 추가
- 글쓰기 스크린 레이아웃 완성
- 글쓰기  완료 버튼 클릭 시 작성한 데이터(사진, 제목, 가격, 본문) 전송 준비 완료
- 카테고리 선택 스크린에서 카테고리 선택 시 글쓰기 페이지로 돌아가고 선택한 카테고리로 변경


## 기타 질문 및 특이 사항



## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.
- [x] Git rebase와 squash를 했고 커밋 수가 하나 인것을 확인 했습니다.

dipsiiiiiiiiii commented on src/screens/commons/postItem/PostItem.js:
```diff
@@ -1,13 +1,77 @@
+    navigation.push("CategorySelectPage");
+  };
+
+  const updateImage = (image) => {
+    setImage(image);
+  };
+
+  const updateTilte = (title) => {
+    setTitle(title);
+  };
+
+  const updatePrice = (price) => {
+    setPrice(price);
+  };
+
+  const updateMainText = (mainText) => {
+    setMainText(mainText);
+  };
```
이 정도면 하나로 합칠 수 있지 않을까요?

HTHstudy replied:
지금 작업중인 커밋에는 합쳐놨습니다.
```jsx
  const updateData = (data) => {
    setBody({ ...body, [data.type]: data.value });
  };
```

dipsiiiiiiiiii commented on src/screens/commons/postItem/PostItem.js:
```diff
@@ -1,13 +1,77 @@
+  };
+
+  const handleUploadPhoto = () => {
+    title.length > 4 &&
+    price.length > 2 &&
+    mainText.length > 4 &&
+    image.length > 0
```
이런 조건들은 보기 쉽게 변수 하나에 할당하고 활용해 주세요!

HTHstudy replied:
지금 작업중인 커밋에는 반영되어있습니다.
```jsx
   let isCorrect =
      body.title.length > 0 &&
      body.price.length > 0 &&
      body.description.length > 0 &&
      image.length > 0;
    isCorrect
      ? createFormData(image, body) : alert("nono");
```

dipsiiiiiiiiii replied:
isCorrect 변수에 다시 값을 할당하지 않으니까 const!

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/CategorySelectButton.js:
```diff
@@ -0,0 +1,30 @@
+
+const CategorySelectButton = ({ goCategorySelectPage }) => (
+  <CategoryPressable onPress={goCategorySelectPage}>
+    <CategoryText style={{ fontSize: 17 }}>카테고리 선택</CategoryText>
```
이 inline style은 아래에서도 정의되고 있어서 없어도 되겠네요!

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/MainTextInput.js:
```diff
@@ -0,0 +1,31 @@
+const inputWidth = windowWidth - 32;
+
+const MainTextInput = ({ updateMainText }) => {
+  const [mainText, setMainText] = useState();
```
초기값을 빈 스트링으로 해주시는 게 더 낫겠네요!

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/MainTextInput.js:
```diff
@@ -0,0 +1,31 @@
+  const [mainText, setMainText] = useState();
+
+  useEffect(() => {
+    mainText && updateMainText(mainText);
```
텍스트 값이 빈 값일 때도 업데이트 되어야 하지 않을까요?

HTHstudy replied:
감사합니다!

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/MainTextInput.js:
```diff
@@ -0,0 +1,31 @@
+
+  return (
+    <TextInput
+      style={{
+        minHeight: 120,
+        width: inputWidth,
+        paddingTop: 18,
+        fontSize: 17,
+      }}
```
styled-components 활용해 주세요 :)

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/PriceInput.js:
```diff
@@ -0,0 +1,46 @@
+const inputWidth = windowWidth - 32;
+
+const PriceInput = ({ updatePrice }) => {
+  const [price, setPrice] = useState();
+
+  useEffect(() => {
+    price && updatePrice(price);
+  }, [price]);
```
여기도 마찬가지,
price 초기 값 0으로 해주고 price가 0일 때도 update는 되어야 하지 않을까요!?

dipsiiiiiiiiii commented on src/screens/commons/postItem/components/TitleInput.js:
```diff
@@ -0,0 +1,31 @@
+const inputWidth = windowWidth - 32;
+
+const TitleInput = ({ updateTilte }) => {
+  const [title, setTitle] = useState();
+
+  useEffect(() => {
+    title && updateTilte(title);
+  }, [title]);
```
여기도 앞 리뷰 참고해 주세요!

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다 리뷰 반영해 주세요!

dipsiiiiiiiiii commented on src/screens/commons/postItem/PostItem.js:
```diff
@@ -1,13 +1,99 @@
+  };
+
+  const handleUploadPhoto = () => {
+    let isCorrect =
```
재 할당 하지 않으니까 const!

dipsiiiiiiiiii reviewed (commented):
const로만 바꿔주세요! 질문 답변도 추가로 드릴게요!

dipsiiiiiiiiii commented:
이미지를 formdata에 넣을 때는 하나씩 넣어주시고 타입도 같이 넣어주셔야 합니다.
참고로 아래와 같이 같은 key에 여러 번 append 할 수 있습니다!
참고 링크 : https://stackoverflow.com/questions/51046458/upload-multiple-images-to-server-with-part-params-using-axio-react-native
```
images.forEach((el) => {
  data.append("product_image", { uri:  el.uri.replace("file://", ""), type: "image/jpg"})
})
```

그리고 body에 있는 데이터는 stringify 해서 넣어도 되지만 인코딩 문제가 있을 수 있어서 아래 링크 참고해서 한 번 해보세요!
https://stackoverflow.com/questions/32441963/how-to-use-formdata-in-react-native

Commit:
commit 내역
- Mod: 충돌해결
- Mod: 리뷰 반영 완료
- Mod: API 주소변경
- Add: Custom Alert 추가
- Mod: 충돌 해결
